x/staking/keeper/grpc_query.go

```
@@ -0,0 +1,460 @@
+	}
+
+	return &types.QueryDelegationResponse{DelegationResponse: &delResponse}, nil
+}
+
+// UnbondingDelegation queries unbonding info for give validator delegator pair
+func (k Querier) UnbondingDelegation(c context.Context, req *types.QueryUnbondingDelegationRequest) (*types.QueryUnbondingDelegationResponse, error) {
+	if req == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "empty request")
+	}
+
+	if req.DelegatorAddr.Empty() || req.ValidatorAddr.Empty() {
+		return nil, status.Errorf(codes.InvalidArgument, "Invalid request")
+	}
+	ctx := sdk.UnwrapSDKContext(c)
+	unbond, found := k.GetUnbondingDelegation(ctx, req.DelegatorAddr, req.ValidatorAddr)
+	if !found {
+		return nil, status.Errorf(codes.NotFound, "unbonding delegation with delegator %s not found for"+
+			"validator %s", req.DelegatorAddr, req.ValidatorAddr)
+	}
+
+	return &types.QueryUnbondingDelegationResponse{Unbond: unbond}, nil
+}
+
+// DelegatorDelegations queries all delegations of a give delegator address
+func (k Querier) DelegatorDelegations(c context.Context, req *types.QueryDelegatorDelegationsRequest) (*types.QueryDelegatorDelegationsResponse, error) {
+	if req == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "empty request")
+	}
+
+	if req.DelegatorAddr == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "Invalid request")
+	}
+	var delegations types.Delegations
+	ctx := sdk.UnwrapSDKContext(c)
+
+	store := ctx.KVStore(k.storeKey)
+	delStore := prefix.NewStore(store, types.GetDelegationsKey(req.DelegatorAddr))
+	res, err := query.Paginate(delStore, req.Req, func(key []byte, value []byte) error {
+		delegation, err := types.UnmarshalDelegation(k.cdc, value)
+		if err != nil {
+			return err
+		}
+		delegations = append(delegations, delegation)
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	if delegations == nil {
+		return nil, status.Errorf(
+			codes.NotFound,
+			"unable to find delegations for address %d", req.DelegatorAddr)
+	}
```

ditto, I don't think we should return an error on empty responses